packages/builder/src/stores/builder/screenComponent.ts

```diff
@@ -0,0 +1,80 @@
+import { derived } from "svelte/store"
+import { tables } from "./tables"
+import { selectedScreen } from "./screens"
+import { viewsV2 } from "./viewsV2"
+import { DerivedBudiStore } from "../BudiStore"
+import { findComponentsBySettingsType } from "@/helpers/screen"
+import { Screen, Table, ViewV2 } from "@budibase/types"
+
+interface BuilderScreenComponentStore {}
+
+interface DerivedScreenComponentStore extends BuilderScreenComponentStore {
+  errors: Record<string, string[]>
+}
+
+export class ScreenComponentStore extends DerivedBudiStore<
+  BuilderScreenComponentStore,
+  DerivedScreenComponentStore
+> {
+  constructor() {
+    const makeDerivedStore = () => {
+      return derived(
+        [selectedScreen, tables, viewsV2],
+        ([$selectedScreen, $tables, $viewsV2]): DerivedScreenComponentStore => {
+          const datasources = flattenTablesAndViews($tables.list, $viewsV2.list)
+          return {
+            errors: getInvalidDatasources($selectedScreen, datasources),
+          }
+        }
+      )
+    }
+
+    super({}, makeDerivedStore)
+  }
+}
```

Comment:
NAB: Fair play for making this a full derived BudiStore, but since this is only deriving data (not storing any) and doesn't have any actions, I would probably just make this a native derived store to keep things easier. It might also be better to just name this `componentErrors` or something and make it a `Record<string, string[]>` directly rather than object - again just keeping things simple. `ScreenComponentStore` is a little generic and not obvious what it does, but I know you're probably just thinking ahead of other use cases for it.